File: fix_syntax.js
const fs = require('fs');
const glob = require('glob');

const files = glob.sync('resources/js/Pages/Admin/**/*.vue');

files.forEach(file => {
  let content = fs.readFileSync(file, 'utf8');
  let original = content;

  // Fix the syntax error:
  // router.delete(`/admin/projects/${id
  // })`)
  
  content = content.replace(/router\.delete\(\`\/(admin\/[a-zA-Z\/]+)\/\$\{id\n\s*\)\`\)/g, "router.delete(`/$1/${id}`)");
  
  // also for deleteCat: router.delete(`/admin/skills/categories/${id\n  })`)
  content = content.replace(/router\.delete\(\`\/(admin\/skills\/categories)\/\$\{id\n\s*\)\`\)/g, "router.delete(`/$1/${id}`)");
  
  if (content !== original) {
    fs.writeFileSync(file, content);
    console.log('Fixed syntax in', file);
  }
});

File: fix_syntax.py
import glob
import re

files = glob.glob('resources/js/Pages/Admin/**/*.vue', recursive=True)

for f in files:
    with open(f, 'r') as file:
        content = file.read()
    
    orig = content
    
    # We need to find: router.delete(`/admin/path/${id\n  })`)
    # And replace it with: router.delete(`/admin/path/${id}`)
    
    # In regex: router\.delete\(`(.*?)id\n\s*\)\`\)
    
    content = re.sub(r"router\.delete\(`(.*?\$\{)id\n\s*\)\`\)", r"router.delete(`\1id}`)", content)
    
    if content != orig:
        with open(f, 'w') as file:
            file.write(content)
        print("Fixed syntax in", f)

